### ISSUE TITLE
Unsplash Screensaver fix

### ISSUE THREAD

**Issue by embql** [Lunatixz/KODI_Addons#267]
I set out to fix this showing a black screen instead of photos, as i love the idea of random art that can be curated easily and shown on the TV easily (collections ftw!), and as suspected, Unsplash requires an API key now for users to fetch images. Also there is a rate limit of 50 per hour so i tried to do something about that to not get too many errors and denials.

So first, you have to create an app at Unsplash and get the access key and post the key in the screensaver settings.
As such, I have added that to the settings and changed the endpoints as well as other code to correctly fetch urls (instead of source.unsplash.com I use api.unplash.com etc).

Beware I'm not too good at coding and it can probably be much more efficient, but at least it works - and especially with the collection mode as that is what I use.
Not really sure the other modes are working correctly, but images are returned and displayed so I'm not too invested.

Please take a look at the code and replace settings.xml and gui.py in the folder and subfolders of "Kodi\addons\screensaver.unsplash\resources\" (remember to backup just in case).
If some of this could be used to fix the current code on github feel free to use it. Not too much experience with pull requests so I'm not doing that right now.

gui.py: [https://drive.google.com/file/d/19GMos61...sp=sharing](https://drive.google.com/file/d/19GMos61dMBZ_M3fhVUCorUmB7mWQ3NDn/view?usp=sharing)
settings.xml: [https://drive.google.com/file/d/1oD1ZqtF...sp=sharing](https://drive.google.com/file/d/1oD1ZqtFqhCF4MZYoXsf0USMyXYe6-_Xd/view?usp=sharing)

Hope you can use it, thanks.
best

**Comment by Lunatixz** [comment]
Thanks...

**Comment by aaruni96** [comment]
@embql I get a "not found" error on your files. Can you try reuploading them>

**Comment by embql** [comment]
Of course, [here you go](https://drive.google.com/drive/folders/1PKGTDCKFH2HuGO-CXgGbHnyleni8SRVJ?usp=sharing).

Still works fine for me (at least with cache i guess), but:
- Beware I've seen that it sometimes uses low-res versions (maybe thumbnails) of images, so we'll have to do a minimum resolution skip function at some point at least.
- Also, not sure it uses all images in a collection since I've seen quite a few repeats, but some logging could test that, as well as taking another look at the random/shuffle function. All good things to improve upon :)

best

**Comment by Lunatixz** [comment]
I'll push an update this week to the Kodi repo
